pyproject.toml

```
@@ -20,3 +21,60 @@ duplicate-code,
 import-error,
 unsubscriptable-object
 """
+
+[tool.mypy]
+warn_incomplete_stub = true
+ignore_missing_imports = true
+disallow_untyped_calls = true
+disallow_untyped_defs = true
+disallow_incomplete_defs = true
+check_untyped_defs = true
+disallow_untyped_decorators = true
+warn_redundant_casts = true
+warn_no_return = true
+warn_unreachable = true
+show_error_codes = true
+warn_unused_ignores = true
+
+[tool.poetry]
+name = "slither-analyzer"
+version = "0.9.0"
+license = "AGPL-3.0"
+description="Slither is a Solidity static analysis framework written in Python 3."
+authors = ["Trail of Bits"]
+packages = [
+    { include = "slither" }
+]
+
+[tool.poetry.scripts]
+slither = "slither.__main__:main"
+slither-check-upgradeability = "slither.tools.upgradeability.__main__:main"
+slither-find-paths = "slither.tools.possible_paths.__main__:main"
+slither-simil = "slither.tools.similarity.__main__:main"
+slither-flat = "slither.tools.flattening.__main__:main"
+slither-format = "slither.tools.slither_format.__main__:main"
+slither-check-erc = "slither.tools.erc_conformance.__main__:main"
+slither-check-kspec = "slither.tools.kspec_coverage.__main__:main"
+slither-prop = "slither.tools.properties.__main__:main"
+slither-mutate = "slither.tools.mutator.__main__:main"
+slither-read-storage = "slither.tools.read_storage.__main__:main"
+
+[tool.poetry.dependencies]
+python = "^3.8"
+prettytable = ">=0.7.2"
+pysha3 = ">=1.0.2"
+crytic-compile = { git = "https://github.com/crytic/crytic-compile.git", branch = "master" }
+
+[tool.poetry.dev-dependencies]
```

```suggestion
[tool.poetry.group.dev.dependencies]
```
Then we need to add `poetry install --with ` where the CI has `pip install ".[dev]"